src/main/java/org/opentripplanner/routing/core/BicycleOptimizeType.java

```diff
@@ -9,5 +12,12 @@ public enum BicycleOptimizeType {
   SAFE,
   FLAT,/* needs a rewrite */
   GREENWAYS,
-  TRIANGLE,
+  TRIANGLE;
+
+  /**
+   * Return all values that are not {@link BicycleOptimizeType#TRIANGLE}.
+   */
+  public static Stream<BicycleOptimizeType> nonTriangleValues() {
+    return Arrays.stream(values()).filter(t -> t != BicycleOptimizeType.TRIANGLE);
+  }
```

Comment:
Do not return a Stream of enums, return a `Set< BicycleOptimizeType>`. Create it like this `EnumSet.complementOf(EnumSet.of(TRIANGLE))`. Since the EnumSet is mutable, creating one each time is one option, another option is to store it in a static instance `Collections.unmodifiableSet(EnumSet.complementOf(EnumSet.of(TRIANGLE)))`.